### PR TITLE
Connect MongoDB

### LINKED ISSUE #87: Connect to MongoDB
The purpose of this ticket is to connect our application to MongoDB so we can store our data there!

- [x] You can look at app portal or this [guide](https://hevodata.com/learn/next-js-mongodb-connection/) for reference on how to connect our application to MongoDB 
- [x] For the GET all templates, swap out the logic where we return the mock data to returning data from the database
- [x] To test that the connection works, I think u can add fake data in MongoDB Atlas by clicking insert document 
Screen Shot 2022-09-05 at 11 09 46 PM and then sending a GET all templates request. It should return the templates in the database if the database is connected properly!

YOUR MENTOR for this ticket is the monogdb king @deanframe - ask him if ya'll get stuck at any point / have questions

### PR DESCRIPTION
**Pull request by mike10911**
Establish GET endpoint and MongoDB connection

Fixes #87 

Changelist:
- Added mongo.ts to handle the connection to templates database on MongoDB Atlas
- Removed previous mock templates and data handling logic and replace with GET logic for Mongo

Notes:
- Installed MongoDB
- Installed next-connect

Tested:
- Tested by inserting data in MongoDB Atlas and hitting the endpoint on our local machine

Screen Shot 2022-09-24 at 3 06 30 PM

### CODE REVIEW

File: pages/api/templates.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+      break
+    case 'GET':
+      const allTemplates = await db.collection('templates').find({}).toArray()
+      res.json({ status: 200, data: allTemplates })
+      break
+  }
+}
+
+export async function getServerSideProps (context) {
+  const res = await fetch('http://localhost:3000/api/templates', {
```
Comment (jzhzhang66) [inline pages/api/templates.js:19]: You can change this to just /api/templates! Right now this will only work for localhost but not production

**Review by jzhzhang66 — CHANGES_REQUESTED** [review]
Overall looks fantastic!!!! BACKEND KINGSSSSSS<333 and s/o to @deanframe for being a fantastic mentor!!<3 

Could we also add logic to make sure that people can't access the data when they're not signed in? 

https://user-images.githubusercontent.com/59738880/192120474-5604941d-ec65-442d-9d46-d7dc3de65a72.mov

Also, please make sure to add the .env variable on vercel if u haven't yet! I haven't checked the error but this might be the issue

**Review by jhylisa666 — APPROVED** [review]
LGTMMM!!!!!!! YALLLL SOOOOO SMARTTTTTT

**Comment by jzhzhang66** [comment]
The endpoint can still be accessed when not signed in - here's what a fix would look like (kind of also related to our workshop content heh): 

You get the request session and if there is a session, then show the data. Otherwise, send a 401 status code w / a nice little message that the unauthorized user will receive. 

Screen Shot 2022-09-28 at 10 59 09 PM

**Review by jzhzhang66 — APPROVED** [review]
LGTM

**Pull request merged**